### PR TITLE
Add functional operators article

### PR DESCRIPTION
**Pull request by Ben-Henshall**
Knowing about the codebases functional operators would be useful:

* What they do
* Why we use them
* Common use cases / examples
* Further documentation

### CODE REVIEW

**Comment by Ben-Henshall** [comment]
Added task to write guide: https://babylonpartners.atlassian.net/browse/CNSMR-2693

File: Cookbook/README.md [hunk 1]
```diff
@@ -38,6 +38,7 @@ iOS Cookbook 👩‍🍳
 ##### ... learn functional programming 🚀
 
 * [Introduce "fish" operator (Kleisli composition)](./Proposals/Fish_Operator.md)
+* Learn about our functional operators [Coming soon...]
```
Comment (mr-v) [inline Cookbook/README.md]: Not sure if there's much value in adding "coming soon" item. I guess it helps to know that someone is working on this and direct any queries to you! 😄 I'd add "CNSMR-2693" there, so it's possible to check on the status.

Comment (AnnKatF) [reply Cookbook/README.md]: This is a public repo so I wouldn't add links to private JIRA Tickets. 
I believe we should only add this link once we have the documentation for it.

**Comment by Ben-Henshall** [comment]
I'll use this PR to add the article instead.


Please don't close

**Comment by Ben-Henshall** [comment]
Updated with the actual article

File: Cookbook/Technical-Documents/FunctionalOperators.md [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+
+
+## `^` - Function lifting
+
+The `^` (caret) operator is used to lift values to become functions.
+
+### Form
+There are two overloads for this specific operator:
+
+1. (A) -> () -> B
```
Comment (sdaniel55) [inline Cookbook/Technical-Documents/FunctionalOperators.md]: [nit]
```suggestion
1. `(A) -> () -> B`
```

**Review by danspndl — APPROVED** [review]
Thank you for writing this! Much easier to understand than going through their definitions 😊

**Review by brocoo — APPROVED** [review]
This is great 👍

**Review by mr-v — COMMENTED** [review]
1. Good initiative and content. 👍 We definitely were missing docs for these.

2. A lot of those operators were explained in pointfree episodes. As a bonus it would be nice to add links to those episodes where applicable.

3. I think it would be also great to get feedback from @inamiy on this doc.

File: Cookbook/Technical-Documents/FunctionalOperators.md [hunk 3]
```diff
@@ -0,0 +1,230 @@
...
+3. For a similar reason to pipe forward, readability.
+ We can clearly follow the function names that describe the sequence of transformations that are happening to the input to return our output. Version 2 is a little harder to follow due to having intermediate variables. (Also, more intermediate variables = more chance to accidentally use the wrong value, such as using `currentValue` where `newValue` should have been used.)
+
+### Variants
+
+#### `<<< ` - Backwards composition
+`<<<` is the reversed version of `>>>`. It combines a function from the right-hand side with a function on the left-hand side. This can occasionally help with readability.
+
+#### `<> ` - Single-type composition
+`<>` is composition that applies between two values of the same type. For example, `5 |> incr <> incr <> incr <> incr` would produce 10 and is the same as `5 |> incr >>> incr >>> incr >>> incr` but with more accurate semantics.
```
[the remark this answers is not in the document]
Comment (inamiy) [reply Cookbook/Technical-Documents/FunctionalOperators.md]: `<>` is not only for function composition but also works for any semigroup/monoid structure appending, e.g. `1 <> 2 <> 3 == 6` (can be addition or multiplication).

So, I would say `<>` has a broader meaning, and not worth mentioning in this context.

File: Cookbook/Technical-Documents/FunctionalOperators.md [hunk 4]
```diff
@@ -0,0 +1,230 @@
...
+
+
+## `^` - Function lifting
+
+The `^` (caret) operator is used to lift values to become functions.
+
+### Form
+There are two overloads for this specific operator:
+
+1. `(A) -> () -> B`
```
Comment (inamiy) [inline Cookbook/Technical-Documents/FunctionalOperators.md]:
```suggestion
1. `(A) -> () -> A`
```

File: Cookbook/Technical-Documents/FunctionalOperators.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (inamiy) [reply Cookbook/Technical-Documents/FunctionalOperators.md]: @Ben-Henshall 
Yes, we can just skip mentioning about `<>` (or at least in a separate context to discuss in more detail).

**Comment by Ben-Henshall** [comment]
@mr-v Changes addressed

**Review by mr-v — APPROVED** [review]
👍

**Review by inamiy — APPROVED** [review]

**Pull request merged**